arangod/RocksDBEngine/RocksDBTransactionState.cpp

```diff
@@ -477,12 +272,8 @@ Result RocksDBTransactionState::abortTransaction(transaction::Methods* activeTrx
   TRI_ASSERT(_status == transaction::Status::RUNNING);
   TRI_ASSERT(activeTrx->isMainTransaction());
   
-  Result result;
+  Result result = _rocksMethods->abortTransaction();
 
-  if (_rocksTransaction != nullptr) {
-    rocksdb::Status status = _rocksTransaction->Rollback();
-    result = rocksutils::convertStatus(status);
-  }
   cleanupTransaction();  // deletes trx
```

Comment:
Ah, it seems `cleanupTransaction` is called here (after the abort). How the cleanups are currently done feels a bit awkward.

Reply:
Yes, unfortunately it has been like this before and I wanted to limit the changes to avoid unintended consequences. But I agree that this feels akward - I'll try to clean this up a bit more.